specs/Standards/Sould Bound Token/README.md

```diff
@@ -0,0 +1,232 @@
+
+## Specification
+
+Soulbound tokens need to be recoverable in case a user's private key is compromised (due to extortion, loss, etc). This becomes especially important for proof-of-human stamps and other NFTs that can only be issued once per user.
+
+Two safeguards against misuse of recovery are contemplated. 1) Users cannot recover an SBT by themselves. The issuer, a DAO or a smart contract (eg: multisig) dedicated to manage the recovery should be assigned. 2) Whenever a recovery is triggered then the wallet from which the NFT was recovered gets blacklisted (indicator that the account identity is burned). Recovering one SBT triggers a blacklist that should apply for SBTs that share the same address.
+
+Soulbound tokens with expire date SHOULD have an option to be renewable. Examples include mandatory renewal with some frequency to check that the owner is still alive, or renew membership to a DAO that uses SBTs as membership gating.
+
+### Smart contract interface
+
+The Soulbound Token interface is a subset of NFT, hence the interface follows the [NEP-171](https://github.com/near/NEPs/blob/master/neps/nep-0171.md).
+
+```rust
+/// TokenMetadata defines attributes for each SBT token.
+pub struct TokenMetadata {
+    pub title: Option<String>, // ex. "fist bump with Robert"
+    pub description: Option<String>, // free-form description
+    pub media: Option<String>, // URL to associated media, preferably to decentralized, content-addressed storage
+    pub media_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of content referenced by the `media` field. Required if `media` is included.
+    pub copies: Option<u64>, // number of copies of this set of metadata in existence when token was minted.
+    pub issued_at: Option<u64>, // When token was issued or minted, Unix epoch in milliseconds
+    pub expires_at: Option<u64>, // When token expires, Unix epoch in milliseconds
+    pub starts_at: Option<u64>, // When token starts being valid, Unix epoch in milliseconds
+    pub updated_at: Option<u64>, // When token was last updated, Unix epoch in milliseconds
+    pub extra: Option<String>, // anything extra the SBT wants to store on-chain. Can be stringified JSON.
+    pub reference: Option<String>, // URL to an off-chain JSON file with more info.
+    pub reference_hash: Option<Base64VecU8>, // Base64-encoded sha256 hash of JSON from reference field. Required if `reference` is included.
+}
+
+
+trait SBT {
+    /**********
+    * QUERIES
+    **********/
+
+    /// get the information about specific token ID
+    fn sbt(&self, token_id: TokenId) -> Option<Token>;
+
+    /// returns total amount of tokens minted by this contract
+    fn sbt_total_supply(&self) -> U64;
+
+    /// returns total supply of SBTs for a given owner
+    fn sbt_supply_for_owner(&self, account: AccountId);
+
+    /// Query for sbt tokens
+    fn sbt_tokens(&self, from_index: Option<U64>, limit: Option<u32>) -> Vec<Token>;
```

Comment:
I'm considering removing the optional arguments.
I can't find a use case to have multiple SBTs of the same kind per user (if one user can have multiple SBTs).
So, maybe we can change to single token / user instead of supporting multisets. Currently we are developing SBT for communities and for proof of humanity, and all use cases involve one token per SBT kind / contract. Maybe we can have another standard for Multi-SBTs (a'la ERC-1155) ? But that would be different then what is presented here. MultiSBT could represent a family of POAPs. Each family will have different metadata. Here, SBT contract only represents one "family" so creating multiple families requires a new smart-contract per family (for example every new edition of a meetup / conference etc...).

On the other hand, if that use case exists, then keeping a slice type (list) for results is not really more complex then Option.

Opinions?

Reply:
Agree, I can't think of a use case for multiple of the same. These should be possible to sell or transfer so why would you ever own more than one. The option of having many probably causes more harm than anything (if someone figured out a hack to mint more than one, and sell some of them, while still retaining the original value by keeping one) which we should discourage by making it impossible in the first place and thereby increasing the value and rarity of the one legitimate SBT.

Some thoughts to qualify why: 
-SBTs should function as a proxy for a verifiable credential, and I don't see the need for more than one credential per person or per account. 
- In some use cases it might be possible to mint several SBTs of the same kind per person, but this should only be possible if they are minted into different accounts, all belonging to the same person. 
- If you want to mint multiple POAPs for example to indicate returning to the same event several times, then they sould still be 1-of in my mind, each with a different date or serial stamp (e.g. conference_day1, conference_day2).